pennylane_lightning_gpu/src/simulator/MPIWorker.hpp

```diff
@@ -0,0 +1,284 @@
+#pragma once
+#include <algorithm>
+#include <bit>
+#include <cctype>
+#include <string>
+#include <vector>
+
+#include "MPIManager.hpp"
+#include "cuda_helpers.hpp"
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include <custatevec.h>
+/// @cond DEV
+namespace {
+namespace cuUtil = Pennylane::CUDA::Util;
+using namespace Pennylane::CUDA;
+using namespace Pennylane::Util;
+} // namespace
+/// @endcond
+
+namespace Pennylane::MPI {
+
+enum WireStatus { Default, Target, Control };
+
+/**
+ * @brief Create wire pairs for bit index swap and transform all control and
+ * target wires to local ones.
+ *
+ * @param numLocalQubits Number of local qubits.
+ * @param numTotalQubits Number of total qubits.
+ * @param ctrls Vector of control wires.
+ * @param tgts Vector of target wires.
+ * @return wirePairs Wire pairs to be passed to SV bit index swap worker.
+ */
+inline std::vector<int2> createWirePairs(const int numLocalQubits,
```

Comment:
I'm just thinking, sometimes users like to play with a small wire example, say with 4 qubits, and a small number of processes, say 4 processes. Let's say they try to execute a small circuit with a CCZ gate, will this function work? Are the following constraints:
- gates cannot target more than number of local qubits
- number of global qubits is greater than number of local qubits
? If so, are there guards against them somewhere?

Reply:
Good point. Happy to add some guards. Both this function and the `bitswapindex` API won't work in that scenario. Actually, the single GPU backend will be recommended. In fact, the constrain will be the number of local available qubits (no control/target) should be no less than the number of target or control wires.

Reply:
Yeah, I'm not worried about _sane_ usage, but sometimes when debugging or testing even a developper can adopt edgy settings and then problems start, and you don't know why if your head is somewhere else.